### PR TITLE
An integration test for optional results.

### PR DESCRIPTION
**Pull request by d4rken**
Showed me that my error wasn't with this lib.
See #150

We have lot's of modular tests already, a few more integration tests can't hurt :smile: .

### CODE REVIEW

**Comment by smyrick** [comment]
@d4rken More tests are great, thank you! We have a pattern right now for the test package `com/expedia/graphql/integration/`. Can you more the file there and name it `OptionalResultsTest.kt`

**Comment by d4rken** [comment]
Done. I missed that there was an existing structure for this :sweat_smile:

File: src/test/kotlin/com/expedia/graphql/integration/OptionalResultsTest.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import com.expedia.graphql.TopLevelObject
+import com.expedia.graphql.testSchemaConfig
+import com.expedia.graphql.toSchema
+import graphql.GraphQL
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+
+@Suppress("Detekt.UnusedPrivateMember",
```
Comment (dkuc84) [inline src/test/kotlin/com/expedia/graphql/integration/OptionalResultsTest.kt]: i think some of those suppressions don't apply?

**Comment by smyrick** [comment]
Not sure why it is reporting a drop. LGTM

**Pull request merged**